### ISSUE TITLE
Style/SymbolProc causes problems in auto-correct

### ISSUE THREAD

**Issue by andreasklinger** [rubocop-hq/rubocop#6978]
## Steps to reproduce the problem

```ruby 
class TestClass
  after_transition any => :some_state do |foo|
    foo.bar!
  end
end
```
```rubocop -a test_class.rb
Inspecting 1 file
C

Offenses:

test_class.rb:2:39: C: [Corrected] Style/SymbolProc: Pass &:foo! as an argument to after_transition instead of a block.
  after_transition any => :some_state do |test| ...
                                      ^^^^^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
```
```ruby
class TestClass
  after_transition any => :some_state, &:bar!
end
```
i am not 100% sure if this is actually incorrect transformation or a bug in a related lib - wanted to report to make sure

received this runtime errors when related code ran
```
ArgumentError: wrong number of arguments (given 1, expected 0)
```

## RuboCop version

```
$ rubocop -V
0.67.2 (using Parser 2.6.2.1, running on ruby 2.4.5 x86_64-darwin18)

```

**Comment by koic** [comment]
Yes. `Style/SymbolProc` cop has been marked as unsafe for auto-correct. 
cf. #6913.

A solution is to use `--safe-auto-correct` option.

```console
% rubocop --safe-auto-correct
```

https://github.com/rubocop-hq/rubocop/blob/v0.67.2/manual/auto_correct.md#safe-auto-correct

**Comment by andreasklinger** [comment]
I would not have expected that

I highly recommend making auto-correct safe by default and introduce an
--aggressive-auto-correct
